Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Tools/SMS.hs

```diff
@@ -51,3 +65,68 @@ sendSMS merchantId = Sms.sendSMS handler
       case merchantSmsServiceConfig.serviceConfig of
         DMSC.SmsServiceConfig msc -> pure msc
         _ -> throwError $ InternalError "Unknown Service Config"
+
+data DashboardMessageType = BOOKING | ENDRIDE | ONBOARDING | CASH_COLLECTED deriving (Show, Generic, Eq)
+
+sendDashboardSms ::
+  ( EsqDBReplicaFlow m r,
+    HasCacheConfig r,
+    HasFlowEnv m r '["smsCfg" ::: SmsConfig],
+    EsqDBFlow m r,
+    EsqLocDBFlow m r,
+    EncFlow m r,
+    Redis.HedisFlow m r,
+    CoreMetrics m
+  ) =>
+  Id Merchant ->
+  DashboardMessageType ->
+  Maybe DR.Ride ->
+  Id DP.Person ->
+  Maybe SRB.Booking ->
+  HighPrecMoney ->
+  m ()
+sendDashboardSms merchantId messageType mbRide driverId mbBooking amount = do
+  transporterConfig <- SCT.findByMerchantId merchantId >>= fromMaybeM (TransporterConfigNotFound merchantId.getId)
+  if transporterConfig.enableDashboardSms
+    then do
+      driver <- Esq.runInReplica $ QPerson.findById driverId >>= fromMaybeM (PersonDoesNotExist driverId.getId)
+      smsCfg <- asks (.smsCfg)
+      mobileNumber <- mapM decrypt driver.mobileNumber >>= fromMaybeM (PersonFieldNotPresent "mobileNumber")
+      let countryCode = fromMaybe "+91" driver.mobileCountryCode
+      let phoneNumber = countryCode <> mobileNumber
+          sender = smsCfg.sender
+
+      message <- case mbRide of
+        Just ride -> case mbBooking of
+          Just booking ->
+            if messageType == BOOKING
+              then
+                MessageBuilder.buildSendBookingMessage merchantId $
+                  MessageBuilder.BuildSendBookingMessageReq
+                    { otp = ride.otp,
+                      amount = show booking.estimatedFare
+                    }
+              else do
+                MessageBuilder.buildSendEndRideMessage merchantId $
+                  MessageBuilder.BuildSendEndRideMessageReq
+                    { rideAmount = show amount,
+                      rideId = ride.shortId.getShortId
+                    }
+          Nothing -> pure ""
```

Comment:
If booking is Nothing, it will send an empty message to the user.

Reply:
This case will never arise. ( As we are sending Maybe ride type  so need to handle the Nothing )